theano/sandbox/scan_module/scan_op.py

```diff
@@ -342,10 +342,10 @@ def rval(p=p, i=node_input_storage, o=node_output_storage, n=node):
         return rval
 
     def grad(self, args, g_outs):
-        pass
+        raise NotImplementedError()
 
     def R_op(self, inputs, eval_points):
-        pass
+        raise NotImplementedError()
```

Comment:
I think it's better to keep them as the minimum standard that should be applied by these functions. If someone wants to extend them, they start from here. ScanOp derives from PureOp which has R_op function, so without having R_op here if that gets changed it affects the R_op functionality here, which by default should not be the case, unless they know the affect of what they are causing here.